zipline/utils/events.py

```diff
@@ -109,6 +109,8 @@ def _build_offset(offset, kwargs, default):
     """
     Builds the offset argument for event rules.
     """
+    # Filter down to just kwargs that were actually passed.
+    kwargs = {k: v for k, v in six.iteritems(kwargs) if v is not None}
```

Comment:
What's this change for?

Reply:
Previously, the factory functions on `date_rules` and `time_rules` just took `**kwargs` and forwarded them to the rule, which forwarded to `_build_offset`, which forwarded to `datetime.timedelta`. I changed it so that those explicitly take `(offset, hours, minutes)` (I don't think we've ever documented `offset` working before, but I'm hesitant to just remove it), which means that `kwargs` is always populated now (possibly with `None`), whereas before it was empty.